lib/importer/src/functions.js

```diff
@@ -48,10 +49,88 @@ const importerErrorMappingData = (error, key) => {
+    const session = new session_lib.Session(session_data);
+    const supportedTypes = backend_lib.SessionGetSupportedTypes(session.backendSid);
+    const mappings = backend_lib.SessionGetMappingRules(session.backendSid);
+
+    return Object.entries(mappings).map((entry) => {
+        const columnField = entry[1];
+        if (!columnField) return false; // Skip unmapped columns
+        const columnDef = session.fields.find((field) => field.name == columnField);
+        const columnTypeName = columnDef.type;
+        const columnType = supportedTypes.get(columnTypeName);
+        return {
+            fieldName: columnField,
+            typeName: columnType.displayName
+        };
+    });
+};
+
+//--------------------------------------------------------------------
+// As long as column->field mappings have been set up, this function will return
+// an array indexed on column, with each element being list of possible formats
+// for that column. The array element for a column will be false if the column is mapped to a
+// field whose type does not require formats, otherwise it will be a list of
+// objects with 'name' (internal code), 'displayName' (human-facing name) and
+// 'description' (longer description) fields.
+// --------------------------------------------------------------------
+const importerPossibleColumnFormats = (data) => {
+    // FIXME: This should probably move into the backend, as a SessionSuggestFormats function
+    const session_data = data[IMPORTER_SESSION_KEY];
+    const session = new session_lib.Session(session_data);
+    const supportedTypes = backend_lib.SessionGetSupportedTypes(session.backendSid);
+    const mappings = backend_lib.SessionGetMappingRules(session.backendSid);
+    const hRange = backend_lib.SessionGetHeaderRange(session.backendSid, session.sheet)
+    const fRange = backend_lib.SessionGetFooterRange(session.backendSid, session.sheet)
+    const dataRange = backend_lib.SessionSuggestDataRange(session.backendSid, hRange, fRange);
+    const typeGuesses = backend_lib.SessionGuessTypes(session.backendSid, dataRange);
+
+    return Object.entries(mappings).map((entry) => {
+        const [colIndex, columnField] = entry;
+        if (!columnField) return false; // Skip unmapped columns
+        const columnGuesses = typeGuesses[colIndex];
+        const columnDef = session.fields.find((field) => field.name == columnField);
+        const columnTypeName = columnDef.type;
+        const columnType = supportedTypes.get(columnTypeName);
+        let likelyFormats = new Set(); // Names of formats this column is likely to have, according to the guesser
+        if (columnGuesses.types && columnGuesses.types.has(columnTypeName)) {
+            const guessedFormats = columnGuesses.types.get(columnTypeName);
+            if(guessedFormats && columnType.formats) {
+                likelyFormats = new Set(Array.from(columnType.formats.keys()).filter((fmtName) => {
```

Comment:
Wonder why there isn't a Set.from() ? 

The filter function can have a short form anonymous function, so instead of 

```
.filter((fmtName) => {return guessedFormats.includes(fmtName)});
```

you can do this if there's only one expression..

```
.filter((fmtName) => guessedFormats.includes(fmtName) );
```

Just saves a couple of { } and a return ...